irohad/model/converters/json_common.hpp

```diff
@@ -18,35 +18,336 @@
 #ifndef IROHA_JSON_COMMON_HPP
 #define IROHA_JSON_COMMON_HPP
 
+#include <algorithm>
+#include <string>
+#include <unordered_map>
+
+#define RAPIDJSON_HAS_STDSTRING 1
 #include <rapidjson/document.h>
-#include <initializer_list>
+#include <rapidjson/prettywriter.h>
+#include <rapidjson/stringbuffer.h>
 #include <nonstd/optional.hpp>
-#include <string>
+
+#include "common/byteutils.hpp"
+#include "model/block.hpp"
+#include "model/common.hpp"
 #include "model/signature.hpp"
 
 namespace iroha {
   namespace model {
     namespace converters {
+      /**
+       * Create map get function for value retrieval by key
+       * @tparam K - map key type
+       * @tparam V - map value type
+       * @param map - map for value retrieval
+       * @return function which takes key, returns value if key exists,
+       * nullopt otherwise
+       */
+      template <typename K, typename V>
+      auto makeOptionalGet(std::unordered_map<K, V> map) {
```

Comment:
This method is pretty general, so you can move it to another class, so that it can be used in other methods too.